@jm-mikkelsen can you address the checkstyle failure and update this.

```
error: commit message body contains line over 72 characters
make: *** [Makefile:15005: commitcheck] Error 1
make: *** Waiting for unfinished jobs....
Error: Process completed with exit code 2.
```

--- a/module/os/freebsd/zfs/zfs_vnops_os.c
+++ b/module/os/freebsd/zfs/zfs_vnops_os.c
@@ -3519,7 +3519,7 @@ zfs_do_rename_impl(vnode_t *sdvp, vnode_t **svpp, struct componentname *scnp,
 				    ZRENAMING, NULL));
 			}
 		}
-		if (error == 0) {
+		if (error == 0 && zfsvfs->z_use_namecache) {
 			cache_vop_rename(sdvp, *svpp, tdvp, *tvpp, scnp, tcnp);
 		}
 	}

